src/main/java/com/gooddata/md/MetadataService.java

```diff
@@ -479,4 +482,18 @@ private IdentifiersAndUris getUrisForIdentifiers(final Project project, final Co
         }
         return response;
     }
+
+    private <T extends Queryable> String getQueryType(final Class<T> cls) {
+        final String typeBase = cls.getSimpleName().toLowerCase();
+
+        if (ReportDefinition.class.isAssignableFrom(cls)) {
```

Comment:
Could you please add test with child class with this fixed (reversed) logic?

Reply:
fixed by changing to equals as we offline agreed